pkg/kanx/kanx_test.go

```diff
@@ -20,12 +22,20 @@ import (
 // Hook up gocheck into the "go test" runner.
 func Test(t *testing.T) { TestingT(t) }
 
+const (
+	ServerReadTimeout = 90 * time.Second
+)
+
 type KanXSuite struct{}
 
 var _ = Suite(&KanXSuite{})
 
 func tmpDir(c *C) string {
-	d, err := os.MkdirTemp("", c.TestName())
+	// unix socket addresses typically cannot be longer than 100 characters
+	// limit the size of the address while retaining some of the properties
+	// of the original directory name
+	hs := adler32.Checksum([]byte(c.TestName()))
```

Comment:
tricky bug from `syscall` package.  Paths to unix sockets can be a maximum of approximately 100 bytes (characters).